pkg/agent/manager/sync.go

```diff
@@ -0,0 +1,302 @@
+	defer m.c.Log.Debug("checkExpiredCacheEntries finished")
+
+	entryRequests := entryRequests{}
+	for entry := range m.cache.Entries() {
+		ttl := entry.SVID.NotAfter.Sub(time.Now())
+		lifetime := entry.SVID.NotAfter.Sub(entry.SVID.NotBefore)
+		// If the cached SVID has a remaining lifetime less than 50%, prepare a
+		// new entryRequest to ask for a new cache entry to be used in the future
+		// when this entry expires.
+		if ttl < lifetime/2 {
+			privateKey, csr, err := m.newCSR(entry.RegistrationEntry.SpiffeId)
+			if err != nil {
+				return nil, err
+			}
+
+			bundles := make(map[string][]byte) //TODO: walmav Populate Bundles
+			cacheEntry := &cache.Entry{
+				RegistrationEntry: entry.RegistrationEntry,
+				SVID:              nil,
+				PrivateKey:        privateKey,
+				Bundles:           bundles,
+			}
+			parentID := entry.RegistrationEntry.ParentId
+			entryRequests[parentID] = append(entryRequests[parentID], &entryRequest{csr, cacheEntry})
+		} else if ttl <= 0 {
+			// Cached SVID expired, remove entry from the cache.
+			m.cache.DeleteEntry(entry.RegistrationEntry)
+		}
+	}
+	return entryRequests, nil
+}
+
+func (m *manager) checkForNewCacheEntries(regEntries map[string]*proto.RegistrationEntry, entryRequests entryRequests) (entryRequests, error) {
+	m.c.Log.Debug("checkForNewCacheEntries started")
+	defer m.c.Log.Debug("checkForNewCacheEntries finished")
+
+	for _, regEntry := range regEntries {
+		if !m.isAlreadyCached(regEntry) {
+			m.c.Log.Debugf("Generating CSR for spiffeId: %s  parentId: %s", regEntry.SpiffeId, regEntry.ParentId)
+
+			privateKey, csr, err := m.newCSR(regEntry.SpiffeId)
+			if err != nil {
+				return nil, err
+			}
+
+			bundles := make(map[string][]byte) //TODO: walmav Populate Bundles
+			cacheEntry := &cache.Entry{
+				RegistrationEntry: regEntry,
+				SVID:              nil,
+				PrivateKey:        privateKey,
+				Bundles:           bundles,
+			}
+			parentID := regEntry.ParentId
+			entryRequests[parentID] = append(entryRequests[parentID], &entryRequest{csr, cacheEntry})
+		} else {
+			m.c.Log.Debugf("cache hit for spiffeId: %s, parentId: %s, selectors: %v", regEntry.SpiffeId, regEntry.ParentId, regEntry.Selectors)
+			// This entry is a cached agent alias, we must synchronize updates on behalf of it.
+			if m.isAgentAlias(regEntry) {
+				err := m.synchronize(regEntry.SpiffeId)
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+	}
+
+	return entryRequests, nil
+}
+
+func (m *manager) rotateSVID() error {
+	m.c.Log.Debug("rotateSVID started")
+	defer m.c.Log.Debug("rotateSVID finished")
+
+	svid, _ := m.getBaseSVIDEntry()
+	ttl := svid.NotAfter.Sub(time.Now())
+	lifetime := svid.NotAfter.Sub(svid.NotBefore)
+
+	if ttl < lifetime/2 {
+		m.c.Log.Debug("Generating new CSR for BaseSVID")
+
+		privateKey, csr, err := m.newCSR(m.spiffeID)
+		if err != nil {
+			return err
+		}
+
+		client := m.getRotationClient()
+
+		m.c.Log.Debug("Sending CSR")
```

Comment:
This feels like a noisy debug line that doesn't provide a lot of useful information

Reply:
You are right. I'll remove it.